### DISCUSSION TITLE
JS SDK - onError refresh

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#10101] [Feature Requests]
Please add a callback function to the sdk what to do in case of an "INVALID_CREDENTIALS" so that we can redirect users to login pages in case of expired or deleted sessions.

Like from: /directus-main/app/src/api.ts line: 57

**Comment by NilsBaumgartner1994** [comment]
Current Solution, implement a Transport wrapper:

import {Transport, TransportMethods, TransportOptions, TransportResponse} from "@directus/sdk";

export default class TransportWrapper extends Transport{

	protected async request<T = any, R = any>(
		method: TransportMethods,
		path: string,
		data?: Record<string, any>,
		options?: Omit<TransportOptions, 'url'>
	): Promise<TransportResponse<T, R>> {
		try {
			return await super.request(method, path, data, options);
		} catch (err: any) {
			console.log("Handle error and check for invalid credentials")

			if (!err || err instanceof Error === false) {
				throw err;
			}
		}
	}

}

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
